src/racer/nameres.rs

```diff
@@ -525,6 +495,34 @@ fn search_scope_headers(point: Point, scopestart: Point, msrc: Src, searchstr: &
     Vec::new().into_iter()
 }
 
+/// Checks if a scope preblock is a function declaration.
+///
+/// TODO: Handle `extern` functions
+fn preblock_is_fn(preblock: &str) -> bool {
+    // Perform simple checks
+    if preblock.starts_with("fn") || preblock.starts_with("pub fn") || preblock.starts_with("const fn") {
```

Comment:
Iterative approach might be cleaner here. _llvm_ should unroll the loop anyhow. Eg

```rust
for case in &["fn", "pub fn", "const fn"] {
    if preblock.starts_with(case) {
        return true;
    }
}
```